src/nnet3/nnet-chain-training.cc

```diff
@@ -135,9 +156,12 @@ void NnetChainTrainer::ProcessOutputs(const NnetChainExample &eg,
 
     computer->AcceptInput(sup.name, &nnet_output_deriv);
 
-    objf_info_[sup.name].UpdateStats(sup.name, opts_.nnet_config.print_interval,
-                                     num_minibatches_processed_++,
+    objf_info_[sup.name + suffix].UpdateStats(sup.name + suffix,
+                                     opts_.nnet_config.print_interval,
+                                     num_minibatches_processed_,
                                      tot_weight, tot_objf, tot_l2_term);
+    if (!is_adversarial_step)
+      num_minibatches_processed_++;
```

Comment:
I think you can remove the if-statement and put this statement at the end of Train().